pytorch_lightning/metrics/functional/accuracy.py

```diff
@@ -18,23 +18,35 @@
 
 
 def _accuracy_update(
-    preds: torch.Tensor, target: torch.Tensor, threshold: float, top_k: Optional[int], subset_accuracy: bool
+    preds: torch.Tensor,
+    target: torch.Tensor,
+    threshold: float,
+    class_weights: Optional[torch.Tensor],
+    top_k: Optional[int],
+    subset_accuracy: bool
 ) -> Tuple[torch.Tensor, torch.Tensor]:
 
     preds, target, mode = _input_format_classification(preds, target, threshold=threshold, top_k=top_k)
 
+    if class_weights is not None:
+        weights = class_weights.to(target.device)[target]
+    else:
+        weights = torch.ones_like(target)
+
     if mode == "multi-label" and top_k:
         raise ValueError("You can not use the `top_k` parameter to calculate accuracy for multi-label inputs.")
 
     if mode == "binary" or (mode == "multi-label" and subset_accuracy):
-        correct = (preds == target).all(dim=1).sum()
-        total = torch.tensor(target.shape[0], device=target.device)
+        correct = (weights * (preds == target).all(dim=1)).sum()
```

Comment:
Weighting should not appear in this case. Subset accuracy means that all elements in each sample must be correctly predicted - there can be no weighting here. And in binary case (1 class) it doesn't make sense anyway